tests/server/randomWord.test.js

```diff
@@ -20,127 +20,134 @@ jest.mock("../../server/middleware", () => {
-        expect(res.text).toEqual(JSON.stringify([data]));
-    })
-    
-    test("Response 200 - standard usage, multiple requirements", async () => {
-        const data1 = [
-            "word1",
-            "word2"
-        ];
-
-        axios.get.mockResolvedValueOnce({data: data1, status: 200});
-        
-        const data2 = [
-            "word"
-        ];
-
-        axios.get.mockResolvedValueOnce({data: data2, status: 200});
-
-        const res = await request.get("/api/word").send({
-            words: [
-                {
-                    numWords: 2,
-                    length: 5
-                },
-                {
-                    numWords: 1,
-                    length: 4
-                }
-            ] 
-        });
-
-        expect(axios.get.mock.calls[0][1]).toEqual({params: {length: 5, number: 2}});
-        expect(axios.get.mock.calls[1][1]).toEqual({params: {length: 4, number: 1}});
-        expect(res.text).toEqual(JSON.stringify([data1, data2]));
-    })
-})
\ No newline at end of file
+describe("Tests for POST at /api/word", () => {
+  const route = "/api/word";
+
+  beforeEach(() => {
+    axios.get.mockClear();
+    jest.spyOn(console, "log").mockImplementation();
+    jest.spyOn(console, "error").mockImplementation();
+  });
+
+  test("200 - Object within 'words' does not contain property 'length'", async () => {
+    const data = [
+      "word1",
+      "word2"
+    ];
+
+    axios.get.mockResolvedValueOnce({ data: data, status: 200 });
+
+    const response = await request.get(route)
+      .send({
+        words: [
+          {
+            numWords: 2
+          }
+        ]
+      });
+
+    expect(axios.get.mock.lastCall[1]).toEqual({ params: { length: -1, number: 2 } });
+    expect(response.text).toBe(JSON.stringify([data]));
+  });
+
+  test("200 - Object within 'words' does not contain property 'numWords'", async () => {
+    const data = [
+      "word1"
+    ];
+
+    axios.get.mockResolvedValueOnce({ data: data, status: 200 });
+
+    const response = await request.get(route)
+      .send({
+        words: [
+          {
+            length: 5
+          }
+        ]
+      });
+
+    expect(axios.get.mock.lastCall[1]).toEqual({ params: { length: 5, number: 1 } });
+    expect(response.text).toBe(JSON.stringify([data]));
+  });
+
+  test("200 - Proper usage with one requirement set", async () => {
+    const data = [
+      "word1",
+      "word2"
+    ];
+
+    axios.get.mockResolvedValueOnce({ data: data, status: 200 });
+
+    const response = await request.get(route)
+      .send({
+        words: [
+          {
+            numWords: 2,
+            length: 5
+          }
+        ]
+      });
+
+    expect(axios.get.mock.lastCall[1]).toEqual({ params: { length: 5, number: 2 } });
+    expect(response.text).toBe(JSON.stringify([data]));
+  });
+
+  test("200 - Proper usage with multiple requirement sets", async () => {
```

Comment:
~~sets~~ set

Reply:
i'd settle for "objects", but "set" doesn't sound right. There's one object with 2 requirement "objects" (or "sets"), which is why I used "sets"

Reply:
Nah I just misread it; thought it was "Proper usage with multiple requirements sets"